add kube-registry-proxy chart to incubator

bacongobbler opened the pull request:
closes #285

Commit:
add kube-registry-proxy chart to incubator

viglesiasce commented:
@k8s-bot ok to test

viglesiasce commented:
Hey @bacongobbler!

Looks like this is a malformed chart. Its missing the _helpers file that provides some of the templates you are using. Its easiest to just use `helm create` then dump your templates into the templates folder. Also dont forget to add a NOTES.txt file in your templates folder that instructs user on how to get started.

bacongobbler commented:
thanks @viglesiasce. I have applied those changes :)

viglesiasce commented on incubator/kube-registry-proxy/values.yaml:
```diff
@@ -0,0 +1,8 @@
+image:
+  repository: gcr.io/google_containers/kube-registry-proxy
+  tag: 0.4
```
This tag was not found for the repo:
```console
  52s   12s     3       {kubelet gke-junkyard-default-pool-0ca72bdc-botz}                                         Warning  FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "kube-registry-proxy" with ErrImagePull: "Tag 0.4 not found in repository gcr.io/google_containers/kube-registry-proxy"
```

bacongobbler replied:
Yeah, someone from the k8s maintainer team has got to push the new release (pinging @roberthbailey from https://github.com/kubernetes/kubernetes/pull/35797)

bacongobbler replied:
Filed https://github.com/kubernetes/kubernetes/issues/38843 for posterity

bacongobbler replied:
This image has been finally pushed to gcr.io:

```
><> d pull gcr.io/google_containers/kube-registry-proxy:0.4
0.4: Pulling from google_containers/kube-registry-proxy
5040bd298390: Pull complete 
d7a91cdb22f0: Pull complete 
9cac4850e5df: Pull complete 
decd9736c768: Pull complete 
2bb4bfc281bb: Pull complete 
Digest: sha256:1040f25a5273de0d72c54865a8efd47e3292de9fb8e5353e3fa76736b854f2da
Status: Downloaded newer image for gcr.io/google_containers/kube-registry-proxy:0.4
```

viglesiasce reviewed (changes requested):
Please add a NOTES.txt file explaining how to use this chart. Also please fix tag version as the chart does not currently install properly.

Can you also move this to the stable folder?

bacongobbler commented:
thanks for the review @viglesiasce. I'll get back to addressing your comments some time early this week now that kube-registry-proxy:0.4 has been pushed out.

lachie83 commented on incubator/kube-registry-proxy/templates/daemon.yaml:
```diff
@@ -0,0 +1,27 @@
+        - name: REGISTRY_PORT
+          value: {{ .Values.registry.port | quote }}
+        ports:
+        - containerPort: 80
```
Should the port be configurable?

bacongobbler replied:
You'd have to refactor the underlying image to change it, so I don't think there's any use in making it configurable.

lachie83 replied:
No worries.

lachie83 commented:
Should we specific a default host? The UX if you don't is the container crashes because `REGISTRY_HOST` is not defined.

Commit:
set default registry.host

lachie83 commented:
Set default registry.host to "gcr.io"

lachie83 commented:
LGTM
